### DISCUSSION TITLE
Actions triggered on a pull request

### DISCUSSION THREAD

**Discussion by ghost** [community/community#45156] [Pull Requests]
### Select Topic Area

Question

### Body

on:
  push:
    branches: 
      - main
    paths-ignore:
      - "README.md"
      - ".github/workflows/codeql.yml"
  pull_request:
    types:
      - opened
      - synchronize 
      - reopened
      - edited 
    branches: 
      - main

I'm having a problem with the workflow, it won't trigger when an additional commit is added to the pull request. What am I doing wrong?

**Comment by jge162** [comment]
It seems that the problem is with the event "pull_request", specifically the types field. It only lists the events that will trigger the workflow when a pull request is created.

If you want the workflow to be triggered when an additional commit is added to an existing pull request, you should also include the "push" event in the "on" field.

```python
on:
  pull_request:
    types:
    - opened
    - synchronize
    - reopened
    - edited
  push:
  branches:
    - main
    ```

**Reply by jge162** [reply]
This way the workflow will be triggered both when a pull request is created and when a new commit is pushed to the pull request.
